pkg/cluster/operation/upgrade.go

```diff
@@ -46,13 +53,33 @@ func Upgrade(
 	components := topo.ComponentsByUpdateOrder()
 	components = FilterComponent(components, roleFilter)
 
+	pdList := make([]string, 0)
```

Comment:
I think we can move L56-L61 to L76 and L120 and change it as

```go
pdClient := api.NewPDClient(topo.(*spec.Specification).GetPDList(), 10*time.Second, tlsCfg)
```

One line is better than 6 lines

Reply:
In that way, we'll be creating PD client multiple times during the upgrade process.